### PR TITLE
I1950 - deliver central or stochastic template, depending on query param

### PR DESCRIPTION
**Pull request by littledice**
Associated PR: https://github.com/vimc/montagu-webmodels/pull/51

But first merge https://github.com/vimc/montagu-api/pull/281

### CODE REVIEW

File: docs/spec/Responsibilities.md [hunk 1]
```diff
@@ -144,6 +144,17 @@ Schema: [`ResponsibilityAndTouchstone.schema.json`](../schemas/ResponsibilityAnd
         }
     }
 
+## GET /modelling-groups/{modelling-group-id}/responsibilities/{touchstone-id}/{scenario-id}/template/
```
Comment (MartinEden) [inline docs/spec/Responsibilities.md]: This doesn't match up with what's implemented.

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/ExpectationsLogic.kt [hunk 2]
```diff
@@ -2,28 +2,26 @@ package org.vaccineimpact.api.app.logic
...
 {
-    override fun getExpectationsForResponsibility(groupId: String, touchstoneVersionId: String, scenarioId: String): Expectations
+    override fun getExpectations(groupId: String, touchstoneVersionId: String, expectationId: Int):
+            Expectations
     {
         modellingGroupRepository.getModellingGroup(groupId) // throws if group does not exist
         touchstoneRepository.touchstoneVersions.get(touchstoneVersionId) // throws if touchstone version does not exist
-        val responsibilityId = responsibilitiesRepository.getResponsibilityId(groupId, touchstoneVersionId, scenarioId)
-        return expectationsRepository.getExpectationsForResponsibility(responsibilityId)
+        return expectationsRepository.getExpectations(expectationId)
```
Comment (MartinEden) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/ExpectationsLogic.kt]: I'm not sure we are currently checking that this expectation belongs to this modelling group and touchstone.

Comment (littledice) [reply src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/ExpectationsLogic.kt]: Just realised you are doing some refactoring to return expectations with responsibilities, so I'll wait for that to go in as will presumably make this check easier.

**Comment by littledice** [comment]
I'd be grateful if you could look over the `ExpectationsLogicTests` class and just check that the tests are correct. The two methods `getExpectationsForResponsibility` and `getResponsibilityWithExpectations` have the same signature and very similar names, which makes that test class very confusing.

**Comment by MartinEden** [comment]
I'll look at this properly tomorrow morning, but: Perhaps we should get rid of `getExpectationsForResponsibility` and just always use `getResponsibilityWithExpectations`? It returns a super-set of the same data, which I guess is slightly inefficient, but not enough to cause us problems I think. Then there is less to reason about and test.

**Comment by littledice** [comment]
Oh, I just realised that `getExpectationsForResponsibility` is totally obsolete. I wrote it specifically for this endpoint, which is now using the new method :woman_facepalming:

**Pull request merged**